pandas/io/formats/format.py

```diff
@@ -1060,6 +1060,11 @@ def get_result_as_array(self):
         def format_values_with(float_format):
             formatter = self._value_formatter(float_format, threshold)
 
```

Comment:
can you do
``na_rep = self.na_rep`` here, then update (and use that value); so the value attached to self is not changed.